src/main/java/org/networkcalculus/dnc/network/server_graph/ServerGraph.java

```diff
@@ -749,6 +757,38 @@ public Flow getFlow(int id) throws Exception {
 		return result;
 	}
 
+	/**
+	 * Finds all the flows that have exactly path p as their path.
+	 *
+	 * @param p
+	 *             The path which all the returned flows have.
+	 * @return Set of flows that have p as their path.
+	 * @throws Exception
+	 *             If the path is null.
+	 */
+	public Set<Flow> getFlowsByPath(Path p) throws Exception {
+
+        // throw a runtime exception if p is null
```

Comment:
As this is a full sentence, please capitalize the first word and end it with a full stop.